sql/maintenance_utils.sql

```diff
@@ -86,16 +97,23 @@ BEGIN
             RAISE EXCEPTION 'nothing to recompress in chunk "%"', chunk_name[array_upper(chunk_name,1)];
         END IF;
     WHEN status = 3 OR status = 9 OR status = 11 THEN
-        PERFORM @extschema@.decompress_chunk(chunk);
-        COMMIT;
-        -- SET LOCAL is only active until end of transaction.
-        -- While we could use SET at the start of the function we do not
-        -- want to bleed out search_path to caller, so we do SET LOCAL
-        -- again after COMMIT
-        SET LOCAL search_path TO pg_catalog, pg_temp;
+        -- first check if there's an index. Might have to use a heuristic to determine if index usage would be efficient,
+        -- or if we'd better fall back to decompressing & recompressing entire chunk
+        SELECT _timescaledb_internal.get_compressed_chunk_index_for_recompression(chunk) INTO STRICT compressed_chunk_index;
```

Comment:
Do you need the index name or would it be enough to have a more generic function like `is_recompress_chunk_segmentwise_supported` here? Such a function would encapsulate the functionality and not publish unnecessary details about the internals.

Reply:
You are right, I don't need the index oid, I just want to verify that the index exists. I will make the suggested change

Reply:
changed it, if you'd like to take another look

Reply:
Actually the changes I made wrt this broke the CI so I'm still working on it 😅

Reply:
The reason I am using this Oid returning function is because this way I could use the existing `invoke_compression_func_remotely` function that is used for compress and decompress, which works with Oids. And that way did not have to write additional functions.

Reply:
Alternatively, I would either write a similar function to `invoke_compression_func_remotely` that works with bool datums, or I would move what is currently `get_compressed_chunk_index_for_recompression` in an internal function. Which option do you think would be best?